src/common/fam_libfabric.cpp

```diff
@@ -604,7 +614,8 @@ int fabric_completion_wait(Fam_Context *famCtx, fi_context *ctx) {
                 throw Fam_Datapath_Exception("Reading from fabric CQ failed");
             }
         }
-    } while (entry.op_context != (void *)ctx);
```

Comment:
Line #602 above: Need to throw Fam_Datapath_Exception only if err.optcontext == (void *) ctx, otherwise update the context internal[1] with FAM_REQ_COMPLETED_WITH_ERR state and save the err.err, err_prov_errno and err.err_data.